chore: remove now obsolete setup.py

janl opened the pull request:
I ran all the referenced commands in README and the GH workflows. None of them actually seem to require `setup.py` any more. Please let me know if I missed anything.

This is part of the STF work for Conda.

janl committed:
chore: remove now obsolete setup.py

hulkoba commented:
Looks like there are some places left 
https://github.com/neighbourhoodie/conda-smithy/blob/refactor/yaml_ruamel/pyproject.toml#L45
https://github.com/neighbourhoodie/conda-smithy/blob/e1afff951c2e0cea5e2f53cbbcfb82a1cfcb55d6/.github/workflows/tests.yml#L43
https://github.com/neighbourhoodie/conda-smithy/blob/249cc7503b09eb24027daa919db85e2317d5e0c9/tests/recipes/click-test-feedstock/recipe/meta.yaml#L19
https://github.com/neighbourhoodie/conda-smithy/blob/5ad3fa3129715bd4b9b4409b1bd4303636f888b8/tests/recipes/noarch_platforms/recipe/meta.yaml#L13

and some at lint_recipe.py and test_lint_recipe

vepadulano committed:
Prevent slowdown in yum due to high file descriptor count

Set the maximum number of file descriptors to 1024 before going through the `yum
install` step of the build, only in case `yum_requirements.txt` is present in
the recipe. This is done to mitigate a bug with old versions of rpm such as the
one shipped with the Centos7 container. See https://bugzilla.redhat.com/show_bug.cgi?id=1537564

beckermr committed:
Update build_steps.sh.tmpl

beckermr committed:
Update build_steps.sh.tmpl

Co-authored-by: Marcel Bargull <mbargull@users.noreply.github.com>

beckermr committed:
Merge pull request #1960 from vepadulano/yum-fix-ulimit

janl committed:
chore: remove setup.py from test preamble

janl committed:
chore: remove setup.py from ignore files

janl committed:
chore: move setup.py install to pip install

I’m unsure about this one as documentation on the dropped arguments
and how to bring them to the new world is sparse. Tests pass locally,
but might be a red herring.

janl committed:
chore: remove setup.py from lint test

Since this is not about the project’s setup.py, somebody needs to
vet that this is the right thing to do.

janl committed:
chore: add news entry for setup.py removal

janl committed:
feat: bump setuptools to >=70

janl committed:
Revert "chore: remove setup.py from lint test"

This reverts commit 49d77568feda2835f58d092ddda8d209ef31fe6b.

janl committed:
Revert "chore: move setup.py install to pip install"

This reverts commit 4dd881130406c09db21c9914364ec5aa2dfbf81c.
